src/liballoc/str.rs

```diff
@@ -2066,9 +2067,39 @@ impl str {
     /// ```
     #[stable(feature = "repeat_str", since = "1.16.0")]
     pub fn repeat(&self, n: usize) -> String {
-        let mut s = String::with_capacity(self.len() * n);
-        s.extend((0..n).map(|_| self));
-        s
+        if n == 0 {
+            return String::new();
+        }
+
+        // n = 2^k + l (2^k > l)
```

Comment:
Nit: `l` looks like `1`. Better use another letter.

Reply:
Better not use single letter identifiers at all...